src/csharp/Microsoft.Spark.E2ETest/IpcTests/BroadcastTests.cs

```diff
@@ -31,12 +32,36 @@ public BroadcastTests(SparkFixture fixture)
             _df = _spark.CreateDataFrame(new[] { "hello", "world" });
         }
 
+        /// <summary>
+        /// Test Broadcast support by using multiple broadcast variables in a UDF with
+        /// encryption enabled.
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V2_3_2)]
+        public void TestMultipleBroadcastWithEncryption()
+        {
+            _spark.SparkContext.GetConf().Set("spark.io.encryption.enabled", "true");
+            var obj1 = new TestBroadcastVariable(1, "first");
+            var obj2 = new TestBroadcastVariable(2, "second");
+            Broadcast<int> bc = _spark.SparkContext.Broadcast(5);
+            Broadcast<TestBroadcastVariable> bc1 = _spark.SparkContext.Broadcast(obj1);
+            Broadcast<TestBroadcastVariable> bc2 = _spark.SparkContext.Broadcast(obj2);
+
+            Func<Column, Column> udf = Udf<string, string>(
+                str => $"{str} {bc1.Value().StringValue} and {bc2.Value().StringValue}");
+
+            var expected = new string[] { "hello first and second", "world first and second" };
+
+            string[] actual = ToStringArray(_df.Select(udf(_df["_1"])));
+            Assert.Equal(expected, actual);
+        }
+
         /// <summary>
         /// Test Broadcast support by using multiple broadcast variables in a UDF.
         /// </summary>
         [Fact]
         public void TestMultipleBroadcastWithoutEncryption()
         {
+            _spark.SparkContext.GetConf().Set("spark.io.encryption.enabled", "false");
```

Comment:
Can we unset this when done in `Dispose`?

Reply:
I'm sorry I did not get that, do you mean to unset this at the end of the test?

Reply:
Or do you mean to create function Dispose that I call at end of each test and unset it there?